pkg/app/piped/executor/cloudrun/cloudrun.go

```diff
@@ -136,3 +134,17 @@ func apply(ctx context.Context, in *executor.Input, cloudProviderName string, cl
 	in.LogPersister.Infof("Successfully created the service %s", sm.Name)
 	return true
 }
+
+func revisionExists(ctx context.Context, in *executor.Input, client provider.Client, revisionName string) (bool, error) {
+	_, err := client.GetRevision(ctx, revisionName)
+	if err == nil {
+		return true, nil
+	}
+
+	if err == provider.ErrRevisionNotFound {
+		return false, nil
+	}
+
+	in.LogPersister.Errorf("Failed while checking the existence of revision %s", revisionName)
```

Comment:
nits, should we print the error here along with revisionName for future debugging?

Reply:
Right. Fixed this.